android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -638,14 +636,9 @@ public void receiveCommand(@NonNull WebView root, String commandId, @Nullable Re
         root.stopLoading();
         break;
       case "postMessage":
-        try {
-          RNCWebView reactWebView = (RNCWebView) root;
-          JSONObject eventInitDict = new JSONObject();
-          eventInitDict.put("data", args.getString(0));
-          reactWebView.evaluateJavascriptWithFallback("document.dispatchEvent(new MessageEvent('message', " + eventInitDict.toString() + "));");
```

Comment:
I wonder if we can instead do the following:
1. `String json = eventInitDict.toString()`
2. Assert that it's `/^{"data":"[^"\n]*"}$/`, perhaps extending the `[^]` contents
3. Use that `json` string